crates/experimentation_client/src/lib.rs

```diff
@@ -69,34 +76,98 @@ impl Client {
         }
     }
 
-    pub async fn get_applicable_variant(&self, context: &Value, toss: i8) -> Vec<String> {
-        let experiments: Experiments = self.get_satisfied_experiments(context).await;
+    pub async fn get_applicable_variant(
+        &self,
+        context: &Value,
+        toss: i8,
+    ) -> Result<Vec<String>, String> {
+        let experiments: Experiments = self.get_satisfied_experiments(context).await?;
         let mut variants: Vec<String> = Vec::new();
         for exp in experiments {
             if let Some(v) =
-                self.decide_variant(exp.traffic_percentage, exp.variants, toss)
+                self.decide_variant(exp.traffic_percentage, exp.variants, toss)?
             {
                 variants.push(v.id)
             }
         }
-        variants
+        Ok(variants)
     }
 
-    pub async fn get_satisfied_experiments(&self, context: &Value) -> Experiments {
+    pub async fn get_satisfied_experiments(
+        &self,
+        context: &Value,
+    ) -> Result<Experiments, String> {
         let running_experiments = self.experiments.read().await;
-        running_experiments
+        let filtered_running_experiments = running_experiments
             .iter()
             .filter(|(_, exp)| {
                 jsonlogic::apply(&exp.context, context) == Ok(Value::Bool(true))
             })
             .map(|(_, exp)| exp.clone())
-            .collect::<Experiments>()
+            .collect::<Experiments>();
+
+        if let Some(prefix) = context.get("prefix") {
+            let prefix_list: HashSet<&str> = prefix
+                .as_str()
+                .ok_or_else(|| {
+                    log::error!("Prefix is not a valid string.");
+                    format!("Prefix is not a valid string.")
+                })
+                .map_err_to_string()?
+                .split(",")
```

Comment:
can we expect prefix as array of strings instead of comma separated string(since this is not query params of any api),